### PR TITLE
github: build-test all targets using GitHub actions

### PR DESCRIPTION
**Pull request by blocktrron**
This PR enables build-testing all targets using GitHub actions.

GitHub provides runners for public projects free of charge. This way, we are able to build-test all Targets simultaneously, which allows to have a PR built this way in under 2 hours.

The Goal of this PR is not to remove the existing Jenkins CI, as a lock-in to GitHub is not desirable. However we are not able to gather enough resources to host this functionality ourselves.

Demo runs can be found here: https://github.com/blocktrron/gluon/actions

### CODE REVIEW

File: contrib/actions/generate-actions.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Build
+        run: bash contrib/actions/run-build.sh {target_name}
+      - name: Archive build output
+        uses: actions/upload-artifact@v1
+        with:
+          name: output
+          path: output
+"""
+
+TARGETS= [
```
Comment (mweinelt) [inline contrib/actions/generate-actions.py]: Can we use `make list-targets` instead?

File: contrib/actions/generate-actions.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"ar71xx-generic",
+	"ar71xx-tiny",
+	"ar71xx-nand",
+        "ar71xx-mikrotik",
```
Comment (mweinelt) [inline contrib/actions/generate-actions.py]: Weird indentation.

File: modules [hunk 3]
```diff
@@ -1,6 +1,6 @@
 GLUON_FEEDS='packages routing gluon'
 
-OPENWRT_REPO=https://git.openwrt.org/openwrt/openwrt.git
+OPENWRT_REPO=https://github.com/openwrt/openwrt.git
```
Comment (mweinelt) [inline modules]: This switch should be explained in the commit message.

File: contrib/actions/generate-actions.py (hunk 1 above)
Comment (blocktrron) [reply contrib/actions/generate-actions.py]: Yes. Altered so the script accepts the input from `make list-targets`

File: contrib/actions/generate-actions.py (hunk 2 above)
Comment (blocktrron) [reply contrib/actions/generate-actions.py]: It's obsolete now anyways.

File: modules (hunk 3 above)
Comment (blocktrron) [reply modules]: Done

File: contrib/actions/generate-actions.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import sys
+
+ACTIONS_HEAD = """
```
Comment (mweinelt) [inline contrib/actions/generate-actions.py]:
```
# Update this file after adding/removing/renaming a target by running
#  `make list-targets | ./contrib/actions/generate-actions.py > .github/workflows/build-gluon.yml`
```

Comment (blocktrron) [reply contrib/actions/generate-actions.py:5]: fixed

**Comment by mweinelt** [comment]
I guess there is no way to spawn jobs from `make list-targets` dynamically at runtime, no?

**Comment by blocktrron** [comment]
No, jobs cannot spawn other jobs.

We could build all targets in one job, but this way we would lose concurrency.

File: .github/workflows/build-gluon.yml [hunk 5]
```diff
@@ -0,0 +1,310 @@
+
+# Update this file after adding/removing/renaming a target by running
+# `make list-targets BROKEN=1 | ./contrib/actions/generate-actions.py > .github/workflows/build-gluon.yml`
```
Comment (mweinelt) [inline .github/workflows/build-gluon.yml:3]: To be consistent:

```suggestion
# `make list-targets BROKEN=1 | ./contrib/actions/generate-actions.py > ./.github/workflows/build-gluon.yml`
```

**Comment by mweinelt** [comment]
LGTM. Thanks!

**Pull request merged**